test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -1112,7 +1116,22 @@ protected static void wipeAllIndices(boolean preserveSecurityIndices) throws IOE
             }
             final Request deleteRequest = new Request("DELETE", Strings.collectionToCommaDelimitedString(indexPatterns));
             deleteRequest.addParameter("expand_wildcards", "open,closed,hidden");
-            deleteRequest.setOptions(deleteRequest.getOptions().toBuilder().setWarningsHandler(ignoreAsyncSearchWarning()).build());
+
+            // If system index warning, ignore but log
+            // See: https://github.com/elastic/elasticsearch/issues/117099
+            // and: https://github.com/elastic/elasticsearch/issues/115809
+            deleteRequest.setOptions(RequestOptions.DEFAULT.toBuilder().setWarningsHandler(warnings -> {
+                for (String warning : warnings) {
+                    if (warning.startsWith("this request accesses system indices:")) {
```

Comment:
I wonder if shutting down warnings about *all* system indexes is going a bit too far here. Maybe having an list of allowed ones would be better? Also, maybe there's a way to clean up that before we get here? For async unfortunately there isn't since it can be created by multiple APIs which don't have cleanup functions right now, but not sure what the status with watches.

Reply:
Thanks for taking a look @smalyshev! To follow up your points:

> I wonder if shutting down warnings about all system indexes is going a bit too far here

I did wonder that too but I also was struggling to think when failing the test due to a warning on cleanup would be helpful information... The request succeeded and the index is deleted, the warning is a known side effect of what we are doing here so this just feels like noise...

> Maybe having an list of allowed ones would be better?

That's possible but difficult in our case. `.triggered-watches` is a data stream (I think? Or at least ILM managed) so the actual index name in the error will change over time. We could do some pattern matching on the warning text but that starts to feel quite fragile...

> maybe there's a way to clean up that before we get here? For async unfortunately there isn't

Alas this is one of those async cases... Watches can run in the background between tests and cause this index to appear and need cleaning up at any time.


I'm happy to add some more complex pattern matching to this but I would be interested to know the problem it would be solving (Not saying there isn't one, More likely I'm just to new here to spot it yet 😃).

Reply:
Reply by @smalyshev via DM:

```19:29
Stas Malyshev

Two notes there: 1. I'd like to keep the warnings handler code separate if possible, code with multiple levels of lambdas nested into each other is kinda hard to read, imho

2. Even if .triggered-watches is a data stream, we still could scan the list and use startsWith to check for specific warnings, not? If it's too hard though then I think it's ok to ignore all system indices there probably```

Reply:
I have moved the warning handler to it's own method to stop the nested lambdas.

The problem with checking which index threw the warning is that it's part way through the warning message so there would need to be code to extract it, then do the starts with.

I feel that with the fragility of such string parsing code (Especially when we have no guarantees on the future consistency of that warning message) and the probability of other indices cropping up in the future that need ignoring, IT's best to just deal with this once and be done with it.

That said, if anyone can come up with a downside to muting all warnings during this cleanup, I'm happy to do a u-turn and get that ignore list function in place.